src/thor/bidirectional_astar.cc

```diff
@@ -780,24 +861,23 @@ bool BidirectionalAStar::SetReverseConnection(GraphReader& graphreader, const BD
     c = rev_pred.cost().cost + oppcost + fwd_pred.transition_cost().cost;
   }
 
-  // Keep the best ones at the front all others to the back
-  best_connections_.emplace_back(CandidateConnection{fwd_edge_id, rev_pred.edgeid(), c});
-
-  if (c < best_connections_.front().cost)
-    std::swap(best_connections_.front(), best_connections_.back());
-
   // Set thresholds to extend search
-  if (cost_threshold_ == std::numeric_limits<float>::max()) {
-    float sortcost = std::max(rev_pred.sortcost(), fwd_pred.sortcost() + cost_diff_);
+  if (cost_threshold_ == std::numeric_limits<float>::max() || c < best_connections_.front().cost) {
     if (desired_paths_count_ == 1) {
-      cost_threshold_ = sortcost + kThresholdDelta;
+      cost_threshold_ = c + kThresholdDelta;
     } else {
-      cost_threshold_ = sortcost + std::max(kAlternativeCostExtend * sortcost, kThresholdDelta);
+      cost_threshold_ = (1.f + kAlternativeCostExtend) * c + kThresholdDelta;
```

Comment:
why not just set the constant to 1.1 and explain its a multiplier in the comment for it?

Reply:
also i notice the semantic difference of always adding the thresholdDelta rather than using it only if its greater than the sortcost. maybe this is worth a description? i guess its because now we dont double the sortcost we use the actual cost of both sides of the route added together, scale that by 1.1 to do the extension and then we add some fluff (delta) just in case the route is super short and the scaling doesnt do much. anyway a comment to that effect would make this less magic. i know we have some above but i feel like they dont capture the full picture

Reply:
yeah, for short routes scale `1.1` is not enough to find alternatives. Ok, will add a comment about that